lib/cargo.js

```diff
@@ -372,17 +372,19 @@ export function provideBuilder() {
 
       // This function is called before every build. It finds the closest
       // Cargo.toml file in the path and uses its directory as working.
-      const preBuildFunction = multiCrateProjects && function () {
-        const editor = atom.workspace.getActiveTextEditor();
-        this.cwd = undefined;
-        if (editor && editor.getPath()) {
-          const wdInfo = findCargoProjectDir(editor.getPath());
-          if (wdInfo) {
-            if (!wdInfo.root) {
-              const p = path.parse(wdInfo.dir);
-              atom.notifications.addInfo('Building ' + p.base + '...');
+      const preBuildFunction = function () {
+        if (multiCrateProjects) {
```

Comment:
We sometimes need `buildWorkDir` to be set regardless of the `Multi-crate projects` setting. We use it later to create a path for the source code link. As the result of this change, for a single-crate projects `buildWorkDir` will always be equal to the project's root directory.

Setting a value to `buildWorkDir` on every build is the only effect of this change. `this.cwd` is always set before the function call, so the `if (!this.cwd...)` will have no effect with the default settings.